packages/nextra-theme-docs/src/components/head.tsx

```diff
@@ -12,38 +11,41 @@ export function Head(): ReactElement {
   const mounted = useMounted()
 
   return (
-    <NextHead>
-      <title>{config.title + renderString(config.titleSuffix)}</title>
-      {renderComponent(config.head)}
-      {config.unstable_faviconGlyph ? (
-        <link
-          rel="icon"
-          href={`data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'><text x='50' y='.9em' font-size='90' text-anchor='middle'>${config.unstable_faviconGlyph}</text><style>text{font-family:system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";fill:black}@media(prefers-color-scheme:dark){text{fill:white}}</style></svg>`}
-        />
-      ) : null}
-      {mounted ? (
-        <meta
-          name="theme-color"
-          content={renderedTheme === 'dark' ? '#111' : '#fff'}
-        />
-      ) : (
-        <>
-          <meta
-            name="theme-color"
-            content="#ffffff"
-            media="(prefers-color-scheme: light)"
+    <>
+      <NextHead>
+        <title>{config.title + renderString(config.titleSuffix)}</title>
+        {config.unstable_faviconGlyph ? (
+          <link
+            rel="icon"
+            href={`data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'><text x='50' y='.9em' font-size='90' text-anchor='middle'>${config.unstable_faviconGlyph}</text><style>text{font-family:system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";fill:black}@media(prefers-color-scheme:dark){text{fill:white}}</style></svg>`}
           />
+        ) : null}
+        {mounted ? (
           <meta
             name="theme-color"
-            content="#111111"
-            media="(prefers-color-scheme: dark)"
+            content={renderedTheme === 'dark' ? '#111' : '#fff'}
           />
-        </>
-      )}
-      <meta
-        name="viewport"
-        content="width=device-width, initial-scale=1.0, viewport-fit=cover"
-      />
-    </NextHead>
+        ) : (
+          <>
+            <meta
+              name="theme-color"
+              content="#ffffff"
+              media="(prefers-color-scheme: light)"
+            />
+            <meta
+              name="theme-color"
+              content="#111111"
+              media="(prefers-color-scheme: dark)"
+            />
+          </>
+        )}
+        <meta
+          name="viewport"
+          content="width=device-width, initial-scale=1.0, viewport-fit=cover"
+        />
+      </NextHead>
+      {/* since in NextHead we can't use hooks */}
+      {renderComponent(config.head)}
```

Comment:
My original idea was to do this:

```suggestion
        {config.head()}
      </NextHead>
```

So it's required to be a functional component.

Reply:
nice, now it's works 💪. config and router no longer empty
<img width="604" alt="image" src="https://user-images.githubusercontent.com/7361780/185602579-c5ab475b-1175-494b-be00-10395f092af7.png">

Reply:
Thanks!!